plio/tests.py

```diff
@@ -268,3 +273,79 @@ def test_each_image_has_unique_name(self):
         random.seed(10)
         test_image_2 = Image.objects.create()
         self.assertNotEqual(test_image_1.url.name, test_image_2.url.name)
+
+
+class PlioDownloadTestCase(BaseTestCase):
+    def setUp(self):
+        super().setUp()
+        # seed a video
+        self.video = Video.objects.create(
+            title="Video 1", url="https://www.youtube.com/watch?v=vnISjBbrMUM"
+        )
+        # seed a plio
+        self.plio = Plio.objects.create(
+            name="Plio 1", video=self.video, created_by=self.user, status="published"
+        )
+        # seed some sessions for the plio
+        Session.objects.create(plio=self.plio, user=self.user)
+        Session.objects.create(plio=self.plio, user=self.user)
+
+    def test_draft_plio_data_cannot_be_downloaded(self):
+        # change plio status to draft
+        self.plio.status = "draft"
+        self.plio.save()
+        # download plio data
+        response = self.client.get(f"/api/v1/plios/{self.plio.uuid}/download_data/")
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_published_plio_data_can_be_downloaded(self):
+        # download plio data
+        response = self.client.get(f"/api/v1/plios/{self.plio.uuid}/download_data/")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertTrue(isinstance(response, FileResponse))
+
+    def test_non_plio_owner_cannot_download_data(self):
+        # create new user
+        new_user = User.objects.create(mobile="+919988776655")
```

Comment:
have created a `self.user_2` in my recent change in the `BaseTestCase` merged to PR so that you don't have to create a new user again.